src/Bicep.LangServer/Deploy/DeploymentHelper.cs

```diff
@@ -0,0 +1,121 @@
+using Azure.ResourceManager;
+using Azure.ResourceManager.Resources;
+using Azure.ResourceManager.Resources.Models;
+using Bicep.Core;
+using Bicep.Core.Json;
+
+namespace Bicep.LanguageServer.Deploy
+{
+    public class DeploymentHelper
+    {
+        /// <summary>
+        /// Creates a deployment at provided target scope and returns deployment succeeded/failed message.
+        /// </summary>
+        /// <param name="deploymentCollectionProvider">deployment collection provider</param>
+        /// <param name="armClient">arm client</param>
+        /// <param name="template">template used in deployment</param>
+        /// <param name="parameterFilePath">path to parameter file used in deployment</param>
+        /// <param name="id">id string to create the ResourceIdentifier from</param>
+        /// <param name="scope">target scope</param>
+        /// <param name="location">location to store the deployment data</param>
+        /// <returns>deployment succeeded/failed message</returns>
+        public static async Task<string> CreateDeployment(
+            IDeploymentCollectionProvider deploymentCollectionProvider,
+            ArmClient armClient,
+            string template,
+            string parameterFilePath,
+            string id,
+            string scope,
+            string location)
+        {
+            if ((scope == LanguageConstants.TargetScopeTypeSubscription ||
+                scope == LanguageConstants.TargetScopeTypeManagementGroup) &&
+                string.IsNullOrWhiteSpace(location))
+            {
+                return LangServerResources.MissingLocationDeploymentFailedMessage;
+            }
+
+            DeploymentCollection? deploymentCollection;
+            var resourceIdentifier = new ResourceIdentifier(id);
+
+            try
+            {
+                deploymentCollection = deploymentCollectionProvider.GetDeploymentCollection(armClient, resourceIdentifier, scope);
+            }
+            catch (Exception e)
+            {
+                return string.Format(LangServerResources.DeploymentFailedWithExceptionMessage, e.Message);
+            }
+
+            if (deploymentCollection is not null)
+            {
+                JsonElement parameters;
+
+                try
+                {
+                    parameters = GetParameters(parameterFilePath);
+                }
+                catch (Exception e)
+                {
+                    return e.Message;
+                }
+
+                var deploymentProperties = new DeploymentProperties(DeploymentMode.Incremental)
+                {
+                    Template = JsonDocument.Parse(template).RootElement,
+                    Parameters = parameters
+                };
+                var input = new DeploymentInput(deploymentProperties)
+                {
+                    Location = location,
+                };
+
+                string deployment = "deployment_" + DateTime.UtcNow.ToString("yyyyMMddHHmmss");
```

Comment:
Wonder if we should add "bicep" or "vscode" to the deployment name...?

Reply:
Do we have something in the agent string for that?

Reply:
Good idea. Added bicep prefix. It'll make it easier to search in portal:
https://github.com/Azure/bicep/commit/1065b9654697ff74c2d8786a3252c97e0cd01578

Reply:
Added user agent string to https://github.com/Azure/bicep/issues/6160